adam/adam_6217.py

```diff
@@ -0,0 +1,635 @@
+        return self.status_dict_1[self.analog_input_statuses[0]]
+
+    def read_Status_1(self):
+        return self.status_dict_1[self.analog_input_statuses[2]]
+
+    def read_Status_2(self):
+        return self.status_dict_1[self.analog_input_statuses[4]]
+
+    def read_Status_3(self):
+        return self.status_dict_1[self.analog_input_statuses[6]]
+    
+    def read_Status_4(self):
+        return self.status_dict_1[self.analog_input_statuses[8]]
+
+    def read_Status_5(self):
+        return self.status_dict_1[self.analog_input_statuses[10]]
+
+    def read_Status_6(self):
+        return self.status_dict_1[self.analog_input_statuses[12]]
+    
+    def read_Status_7(self):
+        return self.status_dict_1[self.analog_input_statuses[14]]
+
+    # --------------------
+    # Additional methods
+    # --------------------
+
+    def encode_value(self, value, channel):
+        type_code = self.analog_output_types[channel]
+        if   type_code == int("0182", 16): tmp = 0.02  * value/65535.0
+        elif type_code == int("0180", 16): tmp = (0.016 * value/65535.0) + 0.004
+        elif type_code == int("0148", 16): tmp = 10    * value/65535.0
+        elif type_code == int("0147", 16): tmp = 5     * value/65535.0
+        elif type_code == int("0143", 16): tmp = (20   * value/65535.0) - 10
+        elif type_code == int("0142", 16): tmp = (10   * value/65535.0) - 5
+        elif type_code == int("0140", 16): tmp = (2    * value/65535.0) - 1
+        elif type_code == int("0104", 16): tmp =         value/65535.0  - 0.5
+        elif type_code == int("0103", 16): tmp = (0.3  * value/65535.0) - 0.15
+        elif type_code == int("0181", 16): tmp = (0.04 * value/65535.0) - 0.02
+        else: tmp = self.analog_output_values[channel]
+        return tmp
+
+    def decode_type_code(self, channel):
+        return self.code_to_type_dict[self.analog_output_types[channel]]
+
+    def encode_type_code(self, value="0-20mA"):
+        return self.type_to_code_dict[value]
+
+    # --------
+    # Commands
+    # --------
+
+    @command
+    @DebugIt()
+    def ConnectWithDevice(self):
+        try:
+            self.connected_ADAM = ModbusTcpClient(self.DeviceAddress,
+                                                  port=int(502))
+        except ModbusException as e:
+            self.set_state(DevState.FAULT)
+            self.set_status("Modbus exception caught while"
+                            " connecting to device: \n%s" % e)
+        except Exception as e:
+            self.set_state(DevState.FAULT)
+            self.set_status("Exception caught while connecting to device:"
+                            + "\n%s" % e)
+        self.set_state(DevState.ON)
+        self.set_status("Connected do device with IP: "
+                        + str(self.DeviceAddress))
+        self.info_stream("Connected do device with IP: " + str(
+             self.DeviceAddress))
+
+    @command(dtype_in = int, doc_in='Channel number')
```

Comment:
The equal sign shouldn't be surrounded with spaces when you're setting a default argument to a method.